Apps/VisionOS/VisionOS/ViewUtils/CodeSnippetHelper.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+extension String {
+    func escapeQuotes() -> String {
+        return self.replacingOccurrences(of: "\"", with: "\"\"")
+    }
+}
+
+func propertiesToTutorialString(_ properties: [Property],
+                                linePrefix: String = "") -> String {
+    let senatizedProperties = properties.filter { p in
+        !p.name.isEmpty && !p.value.isEmpty
+    }.map { p in
+        Property(
+            name: p.name
+                .trimmingCharacters(in: .whitespacesAndNewlines)
+                .escapeQuotes(),
+            value: p.value
+                .trimmingCharacters(in: .whitespacesAndNewlines)
+                .escapeQuotes())
+    }
+    if (senatizedProperties.isEmpty) {
+        return ""
+    }
+    
+    var res = "["
+    var sep = ""
+    senatizedProperties.forEach { p in
+        res += """
+        \(sep)"\(p.name)": "\(p.value)"
+        """
+        sep = ", "
+    }
+    res += "]"
+    return res
+}
```

Comment:
As written, this function took a lot of time to review to understand what it's doing. I believe, it's doing quite a lot. 

Could this be refactored with the goal of making it faster and easier to read? I think the function body is doing multiple things which is part of what makes it complex. Splitting this up into multiple function calls could help? 

At a minimum, adding comments explaining a chunk of code could help. I am open to ideas.